clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -1077,6 +1083,10 @@ private void onSuccess(final long currentTimeMs,
                 }
             }
 
+            if (hasError) {
+                onFailedAttempt(currentTimeMs);
```

Comment:
don't we have here the same gap, that an early return would make that we don't get to call onFailedAttempt?

Reply:
ah yes, I forget it would early return due to non-retriable exception.